Plugins/Analysis/Analyze-Amcache.ps1

```diff
@@ -0,0 +1,140 @@
+<#
+
+.SYNOPSIS
+    Plugin-Name: Analyze-Amcache.ps1
+    
+.Description
+    Analyzes recovered amcache hives for all hosts that you have collected data from.
+    There are checks built in to not analyze twice. By default, the plugin will look for 
+    results from the current date. You can specify the analysis date with the
+    $AnalyzeDate parameter. When using the $AnalyzeDate parameter, you must put your
+    date in the format of yyyyMMdd.
+
+    Dependencies
+    AmcacheParser.exe (From Eric Zimmerman's Tools. stored in the Power-Response Bin directory)
+
+.EXAMPLE
+    
+    Power-Response Execution
+
+    For current date Analysis just execute 'run'
+
+    To specify a date
+
+    set AnalyzeDate 20190309
+    run
+
+.NOTES
+    Author: Drew Schmitt
+    Date Created: 4/11/2019
+    Twitter: @5ynax
+    
+    Last Modified By:
+    Last Modified Date:
+    Twitter:
+  
+#>
+
+param (
+
+    [Parameter(Mandatory=$true,Position=0)]
+    [System.Management.Automation.Runspaces.PSSession]$Session,
+
+    [Parameter(Mandatory=$false,Position=1)]
+    [DateTime]$AnalyzeDate= (Get-Date)
+
+    )
+
+process{
+
+    #Verify that 7za executables are located in (Get-PRPath -Bin)
+
+    $7za32 = ("{0}\7za_x86.exe" -f (Get-PRPath -Bin))
+    $7za64 = ("{0}\7za_x64.exe" -f (Get-PRPath -Bin))
+
+    $7z64bitTestPath = Get-Item -Path $7za64 -ErrorAction SilentlyContinue
+    $7z32bitTestPath = Get-Item -Path $7za32 -ErrorAction SilentlyContinue
+
+    if (!$7z64bitTestPath) {
+
+        Throw "64 bit version of 7za.exe not detected in Bin. Place 64bit executable in Bin directory and try again."
+
+    } elseif (!$7z32bitTestPath) {
+
+        Throw "32 bit version of 7za.exe not detected in Bin. Place 32bit executable in Bin directory and try again."
+    }
+
+     #Determine system architecture and select proper 7za.exe and Velociraptor executables
+    try {
+     
+        $Architecture = (Get-WmiObject -Class Win32_OperatingSystem -Property OSArchitecture -ErrorAction Stop).OSArchitecture
+    
+        if ($Architecture -eq "64-bit") {
+
+            $Installexe = $7za64
+
+        } elseif ($Architecture -eq "32-bit") {
+
+            $Installexe = $7za32
+
+        } else {
+        
+            Write-Error ("Unknown system architecture ({0}) detected for {1}. Data was not gathered.)" -f $Architecture, $Session.ComputerName)
+            Continue
+        }
+
+    } catch {
+    
+     Write-Error ("Unable to determine system architecture for {0}. Data was not gathered." -f $Session.ComputerName)
```

Comment:
Since we are determining the architecture of the local system in the `try` block, we don't need to reference `$Session.ComputerName` in the error handling

Reply:
done